pkg/controller/direct/logging/utils.go

```diff
@@ -130,7 +192,16 @@ func convertAPItoKRM_LogMetricLabels(apiLabels []*api.LabelDescriptor) []krm.Log
 			Key:         &apiLabel.Key,
 			ValueType:   &apiLabel.ValueType,
 		}
+		// TODO(acpana): this looks like a bug on the api, as the value should be defaulted already.
+		if *kccLabels[i].ValueType == "" {
+			*kccLabels[i].ValueType = "STRING" // "" defaults to "STRING"
+		}
 	}
+
+	sort.Slice(kccLabels, func(i, j int) bool {
```

Comment:
I like this, and I've noticed the API returning in an order I can't discern.  The one gotcha is that an input KRM object does not round-trip, but I guess that's a function of the API.